experiments/optimizer/GPT2/run.py

```diff
@@ -0,0 +1,334 @@
+import time
+
+import argparse
+import torch
+
+import torch.optim as optim
+from torch.utils.data import DataLoader, IterableDataset
+
+import numpy as np
+import random
+from tqdm import tqdm
+
+from datasets import load_dataset
+from transformers import GPT2TokenizerFast
+from transformers.models.gpt2.modeling_gpt2 import GPT2Attention
+
+from typing import Tuple, Optional, Callable
+from torch.optim.optimizer import Optimizer
+
+from model import make_model, _attn_wrapper, _attn_orig
+from constants import model_name, bos_token, eos_token, pad_token
```

Comment:
These variable name should be changed too. Did you test this script before commit the code?